crates/assemble/src/npm/mod.rs

```diff
@@ -281,5 +253,114 @@ fn camel_case(name: &str, mut upper: bool) -> String {
     w
 }
 
+fn relative_url(scope: &url::Url, package_dir: &path::Path) -> String {
+    assert!(package_dir.is_absolute());
+
+    package_dir
+        .to_str()
+        .filter(|_| scope.scheme() == "file")
+        .and_then(|d| scope.path().strip_prefix(d))
+        .map(|path| &path[1..]) // Trim leading '/', which remains after stripping directory.
+        .map(|path| {
+            let mut relative = path.to_string();
+
+            // Re-attach trailing query & fragment components.
+            if let Some(query) = scope.query() {
+                relative.push('?');
+                relative.push_str(query);
+            }
+
+            if let Some(fragment) = scope.fragment() {
+                relative.push('#');
+                relative.push_str(fragment);
+            }
+            relative
+        })
+        .unwrap_or_else(|| scope.to_string())
+}
+
+fn relative_path(from: &models::Collection, to: &models::Collection) -> String {
+    let from = url::Url::parse(&format!("https://example/{}", from.as_str())).unwrap();
+    let to = url::Url::parse(&format!("https://example/{}", to.as_str())).unwrap();
+    from.make_relative(&to).unwrap()
```

Comment:
That doesn't work because the first component is going to be used as the URL base, which means they'll have different bases and a relative join isn't possible (this will return None).

Something like `file://localhost/{}` might work, I'm not 100%. I used `https://example/` to make it clear that this isn't something that's expected to actually exist, and think `file://` might be more confusing in that way.